### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by Harishs84**
Key should always pass a unique value and not the index to have better performance when a component is re-rendered.

### CODE REVIEW

**Comment by hariKarthic** [comment]
`<div className={styles.card} key={`key-${index}`}> `

Isn't **key-${index}** unique enough? ... This is the most elegant solution if our data doesnt have a dedicated **id** property.

**Comment by Harishs84** [comment]
We should always pass a unique ID as the key to tell react that which item has updated. Basically, react used to update the value using Index. It is same that if we are not passing Index also by default React use to check the diff using key index. And if the key is not unique React will think that all the nodes have got updated and it will re-render all the nodes which is a costlier operation. Either we need to get the value from JSON or we can have a helper func that will generate the unique key.

**Comment by hariKarthic** [comment]
@Harishs84 Got it after the discussion :+1:

File: README.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ const Card = (props) => {
   return (
     <div className={styles.cardContainer}>
       {cardDetails && cardDetails.map((card, index) =>
-        <div className={styles.card} key={`key-${index}`}>
+        <div className={styles.card} key={styles.id}>
```
Comment (whitehorse5353) [inline README.md]: Is it `styles.id` or `card.id` ? since styles are just an object which holds the style property definitions right.

**Comment by whitehorse5353** [comment]
Hi All,
Tried to create a small sample for the same.
[https://www.webpackbin.com/bins/-KhwZ9ozimgy6YA-eNoX](https://www.webpackbin.com/bins/-KhwZ9ozimgy6YA-eNoX)

Change the line `#4` from `key={item.id}` to `key={index}` and inspect the same to experience what @Harishs84 was talking about.

**Comment by Harishs84** [comment]
Ah. Thanks for pointing this @whitehorse5353. The initial commit I have pushed with the card.id, but not sure why the second commit with style.id. Anyway, i have pushed the code with card.id. Thanks once again.

**Pull request merged**